meta/main.yml

```diff
@@ -58,10 +62,18 @@ dependencies:
 
   - role: debops.nginx
     tags: [ 'depend::nginx', 'depend::nginx:gitlab', 'depend-of::gitlab', 'type::dependency' ]
-    nginx_servers: [ '{{ gitlab_nginx_server }}' ]
+    nginx_servers:
+      - '{{ gitlab_nginx_server }}'
+      - '{{ gitlab_nginx_pages_server }}'
     nginx_upstreams: '{{ gitlab_nginx_upstreams }}'
     when: (gitlab_dependencies is defined and gitlab_dependencies)
 
+  - role: debops.nginx
+    tags: [ 'depend::nginx', 'depend::nginx:gitlab', 'depend-of::gitlab', 'type::dependency' ]
+    nginx_servers:
+      - '{{ gitlab_nginx_pages_server }}'
+    when: gitlab_enable_pages
```

Comment:
This makes the `debops.nginx` role and its dependencies executed twice, there's no need for that. Just move the `gitlab_enable_pages` condition to the respective configuration section in `gitlab_nginx_pages_server` variable, you can use the `enabled` parameter to enable/disable a given configuration in `nginx`.